File: pinyougou-sellergoods-interface/src/main/kotlin/com/pinyougou/sellergoods/service/BrandService.kt
package com.pinyougou.sellergoods.service

import com.pinyougou.pojo.TbBrand
import priv.zhong.bean.PageResult

/**
 * @author 钟未鸣
 * @date 2018/3/28
 */
interface BrandService {
    fun findAll(): List<TbBrand>

    /**
     * @param pageNum 当前页数
     * @param pageSize 页面大小
     */
    fun findPage(pageNum:Int,pageSize:Int): PageResult

    fun add(brand :TbBrand)

    fun update(brand: TbBrand)

    fun delete(ids: Array<Long>)

    /**
     * @param brand 查询条件的实体类
     * @param pageNum 当前页数
     * @param pageSize 页面大小
     */
    fun findPage(brand: TbBrand, pageNum:Int,pageSize:Int): PageResult

    fun findBrandList(): List<Map<*,*>>
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService', function ($http) {
    this.findAll = function () {
        return $http.get("../brand/findAll.do");
    };
    this.findPage = function (page, size) {
        return $http.get("../brand/findPage.do?pageNum=" + page + "&pageSize=" + size);
    };
    this.save = function (entity) {
        return $http.post("../brand/add.do", entity);
    };
    this.update = function (entity) {
        return $http.post("../brand/update.do", entity);
    };
    this.dele = function (deleIds) {
        return $http.get("../brand/delete.do?ids=" + deleIds);
    };

    this.search=function(page,size,searchEntity){
        return $http.post("../brand/search.do?pageNum=" + page + "&pageSize=" +
            size, searchEntity);
    };

    //下拉列表数据
    this.selectOptionList=function(){
        return $http.get('../brand/selectOptionList.do');
    }

    this.findBrandList = function(){
        return $http.post("../brand/findBrandList.do")
    }

});